crates/subspace-core-primitives/src/lib.rs

```diff
@@ -55,6 +55,82 @@ pub type Tag = [u8; 8];
 /// Salt used for creating commitment tags for pieces.
 pub type Salt = [u8; 8];
 
+const PUBLIC_KEY_LENGTH: usize = 32;
+
+/// A Ristretto Schnorr public key as bytes produced by `schnorrkel` crate.
+#[derive(Default, Copy, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Encode, Decode, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Debug))]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct PublicKey([u8; PUBLIC_KEY_LENGTH]);
```

Comment:
`PublicKey` and `Signature` look more like crypto primitives to me.

Reply:
Almost, but they offer no methods and just encapsulate byte representation of those. Unfortunately native types do not have SCALE codec impls, so we can't just use those :disappointed:

Reply:
I mean, how about moving them to `crypto.rs`? 😄

Reply:
Doesn't feel more crypto than `[u8; 32]` to me :upside_down_face: